browser/ui/webui/brave_new_tab_message_handler.cc

```diff
@@ -370,8 +346,15 @@ void BraveNewTabMessageHandler::OnStatsChanged() {
   FireWebUIListener("stats-updated", data);
 }
 
-void BraveNewTabMessageHandler::OnPreferencesChanged() {
+void BraveNewTabMessageHandler::OnPreferencesChanged(
+    const std::string& pref_name) {
   PrefService* prefs = profile_->GetPrefs();
+  if (pref_name == kNewTabPageShowBackgroundImage) {
```

Comment:
same as https://github.com/brave/brave-core/pull/4784/files#r387757504
It doesn't make sense to me to have a pref observer in the UI that can just go directly in the service